weather_mv/loader_pipeline/bq.py

```diff
@@ -174,6 +195,49 @@ def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.Lis
                             region=pipeline_options_dict.get('region'))
             logger.info('Region validation completed successfully.')
 
+    def generate_parquet(
+        self,
+        parquet_path: str,
+        lats: t.List,
+        lons: t.List,
+        lat_grid_resolution: float,
+        lon_grid_resolution: float,
+        skip_creating_polygon: bool = False,
+    ):
+        """Generates geo data parquet."""
+        logger.info("Generating geo data parquet ...")
+        # Generate Cartesian product of latitudes and longitudes.
+        lat_lon_pairs = itertools.product(lats, lons)
+        # Create a temp parquet file for writing.
+        with tempfile.NamedTemporaryFile(suffix='.parquet', mode='w+', newline='') as temp:
+            # Define header.
+            header = ['latitude', 'longitude', GEO_POINT_COLUMN, GEO_POLYGON_COLUMN]
+            data = []
+            for lat, lon in lat_lon_pairs:
+                lat = float(lat)
+                lon = float(lon)
+                row = [lat, lon]
+                sanitized_lon = (((lon % 360) + 540) % 360) - 180
+
+                # Fetch the geo point.
+                geo_point = fetch_geo_point(lat, sanitized_lon)
+                row.append(geo_point)
+
+                # Fetch the geo polygon if not skipped.
+                if not skip_creating_polygon:
+                    geo_polygon = fetch_geo_polygon(lat, sanitized_lon, lat_grid_resolution, lon_grid_resolution)
+                    row.append(geo_polygon)
+                else:
+                    row.append(None)
+                data.append(row)
+
+            df = pd.DataFrame(data, columns=header)
+            # Write DataFrame to parquet.
```

Comment:
Sorry, I haven’t been specific enough. Parquet is a cloud optimized format, so I don’t think you need to create a temporary file and then copy it. You should be able to go from memory directly to cloud bucket withthin the pandas dataframe (maybe, at least, via a dask dataframe).

Reply:
You know what? I did some more googling and it seems I’m mistaken. They way folks write parquet to GCS is via gcsfs. That would be a good option, but it is a departure to how we do it so far. So I’m happy either way. Sorry for the trouble!

Reply:
Yeah :), for now, keeping it as it is.